tasks/auth_init.yml

```
@@ -0,0 +1,139 @@
+---
+- name: Check whether admin user is already exist
+  ansible.builtin.command: >
+    mongosh --quiet {{ '--ssl --host ' + mongodb_net_ssl_host if mongodb_net_ssl_mode == 'requireSSL' else '' }} -u {{ mongodb_user_admin_name }} \
+          -p {{ mongodb_user_admin_password }} --host {{ mongodb_net_bindip }} --port {{ mongodb_net_port }} --eval 'db.version()' admin
+  register: mongodb_user_admin_check
+  changed_when: false
+  check_mode: no
+  ignore_errors: true
+  no_log: "{{ disable_logging_for_auth }}"
+
+- name: Include authorization configuration
+  when: mongodb_user_admin_check.rc != 0
+  block:
+    - name: Set default pymongo installation method
+      ansible.builtin.set_fact:
+        mongodb_pymongo_from_pip: true
+      when: mongodb_pymongo_from_pip is not defined
+
+    - name: Force pip to false for known issues (e.g. Debian Buster)
+      ansible.builtin.set_fact:
+        mongodb_pymongo_from_pip: false
+      when:
+        - ansible_distribution == 'Debian'
+        - ansible_distribution_release == 'buster'
+
+    - name: Install pymongo system package (yum/apt)
+      when: not mongodb_pymongo_from_pip
+      block:
+        - name: Install via dnf/yum
+          ansible.builtin.dnf:
+            name: "{{ mongodb_pymongo_package }}"
+            state: present
+            lock_timeout: "{{ yum_lock_timeout | default(30) }}"
+          when: ansible_pkg_mgr == 'yum'
+
+        - name: Install via apt
+          ansible.builtin.apt:
+            name: "{{ mongodb_pymongo_package }}"
+            state: present
+          when: ansible_pkg_mgr == 'apt'
```

Depending on the final logic of this section (venv or host), it might be possible to just use the following code, unless there is something I am unaware of.

```suggestion
      ansible.builtin.package:
        name: "{{ mongodb_pymongo_package }}"
        state: present
```

---

I’ll check it out when I’m working with RHEL 9. I’ll also test it on Ubuntu. If everything works as expected, that’ll be great!